python/regression/PyROOT_regressiontests.py

```diff
@@ -798,6 +798,65 @@ class MyPyClass6345(cppyy.gbl.MyClass6345):
        self.assertEqual(a.bar(), 3)
        self.assertRaises(TypeError, a.bar, 0)
 
+   def test10DeepHierarchyVirtualCall(self):
+       """Virtual call resolution in deep hierarchy (C++->Py->Py)"""
+       # 6470
+
+       cppyy.gbl.gInterpreter.Declare('''
+       class CppBase6470 {
+       public:
+          virtual ~CppBase6470() {}
+          virtual int fun1() const { return 1; }
+          virtual int fun2() const { return fun1(); }
+       };
+       ''')
+
+       CppBase = cppyy.gbl.CppBase6470
+
+       class PyA(CppBase):
+          def fun1(self): return 11
+
+       class PyB(PyA):
+          def fun1(self): return 111
+
+       class PyC(PyA):
+          pass
+
+       base = CppBase()
+       a = PyA()
+       b = PyB()
+       c = PyC()
+
+       self.assertEqual(base.fun2(), 1)
+       self.assertEqual(a.fun2(), 11)
+       self.assertEqual(b.fun2(), 111)
+       self.assertEqual(c.fun2(), 11) # PyA's fun1 should be invoked
+
+   def test11MultiInheritancePyCpp(self):
+       """Multiple inheritance from Python and Cpp classes"""
+
+       class PurePy1:
+          def foo(self): return 1
+
+       class PurePy2:
+          def bar(self): return 2
+
+       cppyy.gbl.gInterpreter.Declare('''
+       class MyCppClass11 {
+       public:
+          int foo() { return 3; }
```

Comment:
What about protected methods? Do they also work as expected? I vaguely remember some discussion here (?)

Reply:
Yes, that works too, it's tested here:
https://github.com/root-project/roottest/blob/master/python/regression/PyROOT_regressiontests.py#L643

Here it does not really matter if it's public or protected, in both cases the MRO is respected (I quickly tried to be 100% sure :D).